File: backend/src/Application/DTO/pipeline_run.py
from dataclasses import dataclass
from datetime import datetime
from typing import Optional
from uuid import UUID


@dataclass(frozen=True)
class PipelineRun:
    """Output DTO from GetPipelineStatus — flattened, wire-friendly view of the
    latest pipeline run for a user. Deliberately lighter than the Domain PipelineRun
    entity: no user_id/pipeline_type/trigger_type, not relevant to an MCP client."""

    id: UUID
    status: str
    current_step: str
    progress: float
    started_at: Optional[datetime]
    finished_at: Optional[datetime]
    error_message: Optional[str]
    step_outcomes: dict[str, str]

File: frontend/src/features/pipeline/types/pipeline_run.ts
export type PipelineStatus =
  | "pending"
  | "running"
  | "completed"
  | "failed"
  | "cancelled";

export type PipelineStep = "collect" | "analyze" | "match" | "done";

export interface PipelineRun {
  id: string;
  user_id: string;
  pipeline_type: string;
  trigger_type: string;
  status: PipelineStatus;
  current_step: PipelineStep;
  progress: number;
  started_at: string | null;
  finished_at: string | null;
  error_message: string | null;
  created_at: string;
}
